### ISSUE TITLE
[VCPKG] vcpkg manifest adds a long delay to builds even when nothing changes

### ISSUE THREAD

**Issue by Cheney-W** [microsoft/vcpkg#14025]
This issue is first reported on [Developer Community](https://developercommunity.visualstudio.com/content/problem/1204028/vcpkg-manifest-adds-a-long-delay-to-builds-even-wh.html?childToView=1215371#comment-1215371).

After enable Use Vcpkg Manifest and adding a vcpkg.json file into a c++ project in Visual Studio, the following steps will be added into the build steps:
Target VcpkgTripletSelection
Target VcpkgInstallManifestDependencies

The Target VcpkgInstallManifestDependencies step often takes more than 3 seconds to detect package changes when building, regardless of any changes to the project. 
For example, just add a space in the cpp file, under normal circumstances the build step will be finished immediately, but now the VcpkgInstallManifestDependencies step will be executed first, and after a few seconds, the compilation and linking steps will be finished immediately.

Customer hopes that the VcpkgInstallManifestDependencies step will be executed only after there are vcpkg-related modifications, and it will be skipped when building general source modifications.

**Comment by pfaco** [comment]
This is happening to me with the following setup:
CMake with presets, where CMAKE_TOOLCHAIN_FILE is set to the vcpkg toolchain.
Whenever cmake runs again - either by calling "cmake --preset=x" or by modifying the CMakeLists file -- vcpkg also runs, even if none of the vcpkg files (vcpkg.json and vcpkg_configuration.json) are modified. I noticed that 90% of vcpkg run time is detecting compiler hash. 

Is there any way to avoid this to happen? Are there plans to add some kind of cache to avoid detecting compiler hash again, or to avoid vpkg running again if its files are not modified?

**Comment by autoantwort** [comment]
#16920 and #17429 are related

**Comment by Jure-BB** [comment]
Is there any workaround for this issue?

**Comment by autoantwort** [comment]
Should not happen anymore if you don't change your vcpkg.json file.

**Comment by Jure-BB** [comment]
For me, vcpkg install runs every time I run CMake configure in VS 2022 (17.2 preview 6). No changes made to vcpkg.json file.

**Comment by autoantwort** [comment]
Yes the issue is only fixed for msbuild (the original issue message is about msbuild). For cmake is will be fixed after https://github.com/microsoft/vcpkg-tool/pull/362 gets merged

**Comment by PatrickKa** [comment]
Any update on this? Recently I had to deal mostly with CMake at work and those extra few seconds on a reconfigure even though nothing vcpkg-related has changed are quite annoying.

**Comment by miss-programgamer** [comment]
Would really enjoy if the priority of this \*checks date\* 5-year old issue were bumped up a smidge.

**Comment by vikhik** [comment]
Bumping this thread, hoping for some action.

**Comment by williamodwyerdb** [comment]
Bumping this as it's causing us big delays in reconfiguration

**Comment by s9w** [comment]
This is literally 81% of our config time. We change packets maybe 3 times per year. Yet this takes minutes, multiple times per day on every dev machine.
